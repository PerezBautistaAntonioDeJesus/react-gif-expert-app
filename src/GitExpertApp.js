
import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GitExpertApp = () => {


    const [categories, setCategories] = useState(['El chavo']);

    return (
    <>
        <h2>Gift Expert App</h2>
        <AddCategory setCategories={setCategories}/>
        <hr />



        <ol>
            { 
                categories.map( (category ) => (
                   <GifGrid 
                        key={ category }
                        category={ category } 
                   />
                ))
            }
        </ol>

            
    </>
  )
}
