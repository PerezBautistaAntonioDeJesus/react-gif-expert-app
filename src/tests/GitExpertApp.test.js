import { shallow } from "enzyme";
import { React } from "react";
import { GitExpertApp } from "../GitExpertApp";


describe('tests on <GifGrid /> component', () => { 
    
    test('should  the component created correctly', () => { 
        
        const wrapper = shallow(<GitExpertApp />);
        expect( wrapper ).toMatchSnapshot();

     });

     test('should have a list of categories', () => { 
         
        const categories = ['El chavo', 'Goku'];

        const wrapper = shallow( <GitExpertApp defaultCategory={ categories } />);

        expect(wrapper).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
      })

 });