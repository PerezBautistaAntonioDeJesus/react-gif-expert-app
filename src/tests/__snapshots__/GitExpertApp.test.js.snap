// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifGrid /> component should  the component created correctly 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`tests on <GifGrid /> component should have a list of categories 1`] = `
<Fragment>
  <h2>
    Gift Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="El chavo"
      key="El chavo"
    />
    <GifGrid
      category="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
